repeat_expansion_variants/README.md

```diff
@@ -0,0 +1,122 @@
+3. Transcript in which the variant resides. In example 1, it would be NM_000044.
+
+Here is the table describing which properties can be extracted from which identifier types:
+
+| Identifier type           | Repeat unit length | Coordinate span | Transcript ID |
+| ------------------------- | ------------------ | --------------- | ------------- |
+| HGVS-like, coding/genomic | sometimes*         | sometimes*      | yes           |
+| HGVS-like, protein        | –                  | –               | –             |
+| Free text                 | yes                | –               | –             |
+
+_\* At least one of (repeat unit length, coordinate span) is always present in this case_
+
+At this step, we extract all properties which we can from the variant identifiers and save them to separate dataframe columns.
+
+### Step 3. Annotate Ensembl gene information
+There are three columns which can be used to deduce gene information: HGNC_ID & GeneSymbol (from ClinVar data), and TranscriptID (parsed from the variant identifier). Because neither of them is present for **all** records, we successively try to annotate Ensembl gene ID based on each of those columns in that order. For example, if Ensembl gene ID for a particular record was determined based on HGNC_ID, then GeneSymbol and TranscriptID will not be tried for that column. Ensembl gene information is being annotated using their [BioMart service](https://www.ensembl.org/biomart/).
+
+In some cases, a mapping from HGNC ID produces multiple Ensembl hits. For example, HGNC:10560 is being resolved to ENSG00000285258 and ENSG00000163635. In this case, a record is further split into two, each mapping into their corresponding Ensembl gene IDs.
+
+### Step 4. Determine repeat type and record completeness
+There are two repeat types detected by this pipeline:
+* `trinucleotide_repeat_expansion`, corresponding to SO term http://sequenceontology.org/browser/current_release/term/SO:0002165. This is chosen when the repeat length appears to be either 3, **or a multiple or 3.** For example:
+  + `NM_001081563.1:c.*224_*226[(50-?)]` — repeat unit spans 3 nucleotides based on coordinates
+  + `NM_001081560.2(DMPK):c.*224_*226CTG(51_?)` — spans 3 nucleotides based on coordinates, and the sequence length is also 3
+  + `NM_000100.3(CSTB):c.-210CCCCGCCCCGCG(2_3)` — sequence length is 12, a multiple of 3
+* `short_tandem_repeat_expansion`, corresponding to SO term http://sequenceontology.org/browser/current_release/term/SO:0002161. This is chosen in the cases when the repeat length can be _determined_ by at least one method, but its unit length is not a multiple of 3.
+
+Similarly to the previous step, repeat type can be determined using several fields. The determination algorithm is as follows:
+
+* If the variant identifier is type 3, “Protein HGVS”: assume repeat is a `trinucleotide_repeat_expansion`, since it affects entire amino acids
```

Comment:
Could you please cite relevant literature that evidences/supports this assumption?

Reply:
There isn't so much literature supporting this, as just the fact that each amino acid is encoded by a triplet of nucleotides. So if we see an insertion of N amino acids, it invariably means an insertion of 3×N nucleotides took place, hence this will be a trinucleotide repeat expansion. I've added this to the description for clarify. Fixed in https://github.com/EBIvariation/vep-mapping-pipeline/pull/3/commits/74339dfa546dfce2a3fa5b355f02f73981895b0b